ethereum/ethevents/events.go

```diff
@@ -290,13 +290,36 @@ func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context, fromBl
 		go ev.runEventProcessor(ctx)
 	}
 
+	// check eth connection
+	connCh := make(chan bool, 1)
+	go func(chan bool) {
+		for {
+			time.Sleep(time.Second * 5)
+			blockTctx, cancel := context.WithTimeout(ctx, types.EthereumReadTimeout*2)
+			if _, err := ev.VotingHandle.EthereumClient.BlockByNumber(blockTctx, nil); err != nil {
```

Comment:
Shouldn't the block number be saved and check that it is actually being increased? Else we won't detect a frozen web3 endpoint

Reply:
there is an else clause that will be executed when restarting the subscription (100 blocks), I think is enough for this purpose. What do you think?

Reply:
will tackle on the next PR, added a better approach :)